torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -1335,17 +1328,18 @@ std::vector<BufPtr> TensorExprKernel::preAllocIntermediateBufs(
     }
     // Only allocate memory for static bufs.
     if (!is_static) {
-      remaining_interm_bufs.push_back(buf);
-      continue;
+      throw std::runtime_error(
+          "TensorExprKernel: preallocation does not support dynamic shapes!");
     }
     auto bp = (void*)malloc(size);
     if (!bp) {
-      remaining_interm_bufs.push_back(buf);
-      continue;
+      throw std::runtime_error(
+          "TensorExprKernel: memory allocation failed during preallocation!");
```

Comment:
Let's use one of TE exceptions instead of generic runtime_error here (see https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/tensorexpr/exceptions.h).